bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,450 @@
+
+As an example, s2i binary build of the https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/mp-config/README.md[microprofile-config example] is provided. 
+
+
+===== Configuring the Maven plugin for Cloud environment
+
+The Maven plugin configuration item ```<cloud></cloud>``` allows to build a bootable JAR for cloud environment. By default the server is configured to run inside an OpenShift context.
+Set the cloud child element ```<type>openshift|kubernetes</type>``` to select the targeted cloud platform.
+
+The sever configuration is updated in order to properly operate in a cloud environment:
+
+* The ```microprofile-health``` and ```core-tools``` (that contains WildFly CLI) galleon layers are provisioned. They are required for the  OpenShift probes and WildFly OpenShift operator to properly operate.
+* The public and private inet addresses are bound to the value of the ```HOSTNAME``` environment variable if defined (defined in OpenShift PODS).
+* The management inet address is bound to the 0.0.0.0 inet address allowing for local (required by WildFly CLI) and remote access (required by OpenShift readiness and liveness probes).
+* The transaction subsystem id is set to the value of ```jboss.node.name```.
+* The ```jboss.node.name``` system propery, if not set, is set to the value of ```HOSTNAME``` environment variable if defined (defined in OpenShift PODS). The node name value
+is truncated to a max of 23 characters in order for the transaction subsystem to properly operate.
+* The server logs are printed in the console.
+* jgroups subsystem is configured to use kubernetes.KUBE_PING jgroups protocol for both tcp (default stack) and udp. PING and MPING protocols are removed.
+* It is possible to configure jgroups to use un-encrypted password authentication. Set the ```<cloud>``` child element ```<enable-jgroups-password>true|false</enable-jgroups-password>``` to enable authentication. 
+NB: When authentication is enabled, the environment variable ```JGROUPS_CLUSTER_PASSWORD``` must be set.
+
+Some examples:
+
+Configure for OpenShift execution:
+
+```
+<cloud/>
+```
+
+Configure for OpenShift execution with jgroups authentication enabled:
+
+```
+<cloud>
+  <enable-jgroups-password>true</enable-jgroups-password>
+</cloud>
+```
+Configure for kubernetes execution:
+
+```
+<cloud>
+  <type>kubernetes</type>
+</cloud>
+```
+
+===== WildFly OpenShift operator
+
+The WildFly OpenShift operator can be used to manage deployments based on image containing a WildFly bootable JAR.
+At boot time, the WildFly bootable JAR dumps in the file ```/opt/jboss/container/wildfly-bootable-jar/install-dir``` its installation path.
+This information is required by the WildFly OpenShift operator to retrieve transaction logs and call into WildFly CLI.
+
+===== JKube Maven plugin
+
+The JKube Maven Plugin is evolved with a generator that recognizes the bootable JAR maven plugin. DETAILS TO BE PROVIDED WHEN FINALIZED.
```

Comment:
IIUC this is going to be released soon, so how is this tracked? Shouldn't it be related to this proposal RFE and maybe listed in the "Related Issues" section?

Reply:
I added the github Issue that tracks adding support to JKube in the "Related Issues"

Reply:
Great, thanks.

Reply:
I updated JKube chapter with more information now that 1.0.0-rc1 is released.

Reply:
Great!